File: Difficulty: Basic/Queue Using Array/queue-using-array.py
class myQueue:
    
    def __init__(self, n):
        # Define Data Structures
        self.arr = [None] * n
        self.capacity = n
        self.front = 0
        self.rear = -1
        self.size = 0

    
    def isEmpty(self):
        # Check if queue is empty
        return True if self.size == 0 else False

    
    def isFull(self):
        # Check if queue is full
        return True if self.size == self.capacity else False

    
    def enqueue(self, x):
        # Enqueue
        if self.isFull():
            return;
        
        self.rear = (self.rear + 1) % self.capacity
        self.arr[self.rear] = x
        self.size += 1

    
    def dequeue(self):
        # Dequeue
        if self.isEmpty():
            return;
        
        self.front = (self.front + 1) % self.capacity
        self.size -= 1

    
    def getFront(self):
        # Get front element
        if self.isEmpty():
            return -1
        
        return self.arr[self.front]
       
    
    def getRear(self):
        # Get rear element 
        if self.isEmpty():
            return -1
        
        return self.arr[self.rear]
        
        

File: Difficulty: Basic/Queue Using Array/queue-using-array.java
class myQueue {
    
    int[] arr;
    int capacity;
    int front;
    int rear;
    int size;
    // Constructor
    public myQueue(int n) {
        // Define Data Structures
        this.arr = new int[n];
        this.capacity = n;
        this.front = 0;
        this.rear = -1;
        this.size = 0;
    }

    public boolean isEmpty() {
        // Check if queue is empty
        return size == 0;
    }

    public boolean isFull() {
        // Check if queue is full
        return size == capacity;
    }

    public void enqueue(int x) {
        // Enqueue
        if(isFull()){
            return;
        }
        rear = (rear+1) % capacity;
        arr[rear] = x;
        size++;
    }

    public void dequeue() {
        // Dequeue
        if(isEmpty()){
            return;
        }
        front = (front+1) % capacity;
        size--;
    }

    public int getFront() {
        // Get front element
        if(isEmpty()){
            return -1;
        }
        return arr[front];
    }

    public int getRear() {
        // Get last element
        if(isEmpty()){
            return -1;
        }
        return arr[rear];
    }
}
